### ISSUE TITLE
Incorrect uncertainty rounding

### ISSUE THREAD

**Issue by HuidaeCho** [josephwright/siunitx#634]
```latex
\documentclass{article}
\usepackage{siunitx}
\begin{document}
\sisetup{
  separate-uncertainty,
  round-mode=uncertainty,
  round-precision=3
}
\begin{tabular}{S}
  123.4567+-12.3456\\
  12.34567+-1.23456\\
  1.234567+-0.123456\\
\end{tabular}
\end{document}
```
produces 123.5+-**123.0**, 12.35+-1.23, and 1.235+-0.123 instead of expected 123+-**12.3**, 12.3+-1.23, and 1.23+-0.123 (if I'm not mistaken). Especially, the first case just multiplies the uncertainty by 10, which makes output incorrect.

![image](https://user-images.githubusercontent.com/7456117/211976249-227b4a1c-ed7e-45a5-a937-a5bdd331cf0f.png)

siunitx was installed from TeXLive 2022.

**Comment by josephwright** [comment]
Working properly for me with release 2023-01-03 v3.2.0; please add `\listfiles` to your input to check your `siunitx` version.

**Comment by HuidaeCho** [comment]
Just updated my TeXLive and **123.0** is now fixed to **12.3**! Thanks for checking.

![image](https://user-images.githubusercontent.com/7456117/212051282-2538a50f-a262-4399-9247-38b30b7b6761.png)

BTW, with a precision of 3 (number of significant figures with uncertainty mode, not places, I guess), I think 123.4567 should be rounded to 123, not 123.5. Similarly, 12.34567 to 12.3, and 1.234567 to 1.23. Based on the above result, it looks like only the uncertainty part is rounded correctly and the main part has one more figure. Is it by design or am I missing something? Is there a way to specify different precisions for both parts? Also, is there a way to use places for the uncertainty round mode?

For our records, this was my file list (siunitx v3.1.9):
```
 *File List*
 article.cls    2022/07/02 v1.4n Standard LaTeX document class
  size10.clo    2022/07/02 v1.4n Standard LaTeX file (size option)
 siunitx.sty    2022-10-23 v3.1.9 A comprehensive (SI) units package
translations.sty    2022/02/05 v1.12 internationalization of LaTeX2e packages (
CN)
etoolbox.sty    2020/10/05 v2.5k e-TeX tools for LaTeX (JAW)
pdftexcmds.sty    2020-06-27 v0.33 Utility functions of pdfTeX for LuaTeX (HO)
infwarerr.sty    2019/12/03 v1.5 Providing info/warning/error messages (HO)
   iftex.sty    2022/02/03 v1.0f TeX engine tests
 ltxcmds.sty    2020-05-10 v1.25 LaTeX kernel commands for general use (HO)
 amstext.sty    2021/08/26 v2.01 AMS text
  amsgen.sty    1999/11/30 v2.0 generic functions
   array.sty    2022/09/04 v2.5g Tabular extension package (FMi)
l3backend-pdftex.def    2022-10-26 L3 backend support: PDF output (pdfTeX)
   color.sty    2022/01/06 v1.3d Standard LaTeX Color (DPC)
   color.cfg    2016/01/02 v1.6 sample color configuration
  pdftex.def    2022/09/22 v1.2b Graphics/color driver for pdftex
supp-pdf.mkii
mathcolor.ltx
translations-basic-dictionary-english.trsl    (english translation file `transl
ations-basic-dictionary')
 ***********
```

**Comment by HuidaeCho** [comment]
> Is it by design

Looking at the result again, I think the main part has the same number of decimal places as the uncertainty part. That makes sense.
